0000-stream/0000-stream.md

```diff
@@ -0,0 +1,407 @@
+---
+title: STREAM - A Multiplexed Money and Data Transport for ILP
+draft: 1
+---
+
+# STREAM: A Multiplexed Money and Data Transport for ILP
+
+## 1. Introduction
+
+STREAM is a multiplexed Interledger Transport Protocol that provides for sending multiple "streams" of money and data between two parties using ILP. STREAM is designed to provide a flexible set of features that allow it to be used for multiple payment and messaging applications:
+- Sending money and data over ILP
+- Segmenting larger payments or messages into packets and reassembling them
+- Bi-directional communication between two endpoints through ILP
+- Stream multiplexing (sending multiple logical streams of money and/or data over a single connection)
+- Stream- and connection-level flow control (adjusting the rate at which money and data are sent)
+- Congestion control (adjusting the rate at which ILP packets are sent based on network throughput limits)
+- Authenticating and encrypting ILP packet data
+- Generating and fulfilling ILP packet conditions
+- Connection migration
+
+STREAM is a successor to the [Pre-Shared Key V2 (PSK2)](../0025-pre-shared-key-2/0025-pre-shared-key-2.md) Transport Protocol and also takes significant inspiration from the [QUIC](https://tools.ietf.org/html/draft-ietf-quic-transport-10) Internet Transport Protocol. Like PSK2, STREAM uses a shared secret to authenticate and encrypt multiple packets, as well as to generate the conditions and fulfillments. In addition, STREAM enables sending money and data in both directions between the two endpoints and automatically determines how much money and data can be sent in each ILP packet. STREAM borrows heavily from QUIC's packet format, stream multiplexing, and approach to flow control (see [Appendix A](#appendix-a-similarities-and-differences-with-quic) for similarities and differences with QUIC).
+
+## 2. Conventions and Definitions
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “NOT RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in BCP 14 [RFC2119](https://tools.ietf.org/html/rfc2119) [RFC8174](https://tools.ietf.org/html/rfc8174) when, and only when, they appear in all capitals, as shown here.
+
+Definitions of terms that are used in this document:
+
+- **Client** - The endpoint initiating a STREAM connection
+- **Server** - The endpoint accepting incoming STREAM connections
+- **Endpoint** - The client or server end of a connection
```

Comment:
I used that term in this rewrite just because that was what QUIC used. Before I was using "entity" or "party" which seem too vague. I hear you on the potential confusion though.

Reply:
I think `Endpoint` is a really good word because it makes me imagine something except for connectors, the very end of the stream which means `sender` or `receiver` even if it sometimes recalls us HTTP endpoint.

To make it better, I propose modifying
- **Endpoint** - The client or server end of a connection
to
- **Endpoint** - The client or server end of a connection, which can be sender or receiver of money and data.